plugins/generator-1.20.1/forge-1.20.1/block.definition.yaml

```diff
@@ -388,6 +388,18 @@ templates:
     writer: json
     name: "@MODASSETSROOT/models/block/@registryname_inventory.json"
 
+block_base_properties:
+  Stairs: [half, shape]
+  Slab: [type]
+  Fence: [north, south, west, east]
+  Wall: [up, north, south, west, east]
+  TrapDoor: [half, open]

```

Comment:
I am fairly confident TrapDoor defines more than just these two:

![image](https://github.com/user-attachments/assets/ab686d85-275c-4762-9d85-234e070a57ec)

Please, go through all block bases again carefully one by one and investigate their source code with extra care to not let those out

Reply:
`facing` and `waterlogged` were included in the `_default` section.

Reply:
Just updated the comment

Reply:
Added missing properties per block base.